src/igvfd/types/file_set.py

```diff
@@ -969,6 +970,30 @@ def externally_hosted(self, request, files=None):
                     externally_hosted_value = True
         return externally_hosted_value
 
+    @calculated_property(
+        schema={
+            'title': 'Software Versions',
+            'description': 'The software versions used to produce this predictive model.',
+            'type': 'string',
+            'linkTo': 'SoftwareVersion',
+            'notSubmittable': True,
+        }
+    )
+    def software_versions(self, request, files=None):
+        software_versions = []
+        if files:
+            for file in files:
+                if file.startswith('/model-files/'):
+                    file_object = request.embed(file, '@@object?skip_calculated=true')
+                    analysis_step_version = file_object.get('analysis_step_version', '')
+                    if analysis_step_version:
+                        analysis_step_version_object = request.embed(
+                            analysis_step_version, '@@object?skip_calculated=true')
+                        software_versions = software_versions + \
+                            analysis_step_version_object.get('software_versions', [])
+        if software_versions:
+            return software_versions
+
```

Comment:
Maybe do` set()` before return and require software_versions has unique items? (e.g. bluestarr has two model files under the same model set)

Reply:
good idea :^)